crypto/evp/p_open.c

```diff
@@ -23,41 +23,44 @@ int EVP_OpenInit(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *type,
                  EVP_PKEY *priv)
 {
     unsigned char *key = NULL;
-    int i, size = 0, ret = 0;
+    size_t keylen = 0;
+    int ret = 0;
+    EVP_PKEY_CTX *pctx = NULL;
 
     if (type) {
         EVP_CIPHER_CTX_reset(ctx);
         if (!EVP_DecryptInit_ex(ctx, type, NULL, NULL, NULL))
-            return 0;
+            goto err;
     }
 
     if (priv == NULL)
         return 1;
 
-    if (EVP_PKEY_id(priv) != EVP_PKEY_RSA) {
```

Comment:
The docs for EVP_SealInit() indicate the following..
The public key must be RSA because it is the only OpenSSL public key algorithm that supports key transport.
Maybe this check for the type is still needed?

Reply:
Ah, I think that documentation needs to be changed. Why limit it to RSA when someone might come up with something new that supports key transport, and writes a provider for it at any given moment in time?

Reply:
Like ECC key transport?

Reply:
Possibly? Doesn't really matter, we should just avoid the of fairly arbitrary limit, especially when libcrypto is becoming a framework.